### PR TITLE
refactor(v2): correct plugin types

### PR DESCRIPTION
**Pull request by armano2**
## Motivation

Improve types in docusaurus, 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### CODE REVIEW

**Comment by armano2** [comment]
Lighthouse Report is failing :(

File: packages/docusaurus/src/server/plugins/init.ts [hunk 1]
```diff
@@ -114,7 +115,7 @@ For more information, visit https://v2.docusaurus.io/docs/using-plugins.`);
         version: pluginVersion,
       };
     })
-    .filter(Boolean);
+    .filter(<T>(item: T): item is Exclude<T, null> => Boolean(item));
```
Comment (slorber) [inline packages/docusaurus/src/server/plugins/init.ts:118]: 🤪

Comment (slorber) [reply packages/docusaurus/src/server/plugins/init.ts:118]: will merge this, but if you can add a utility fn in docusaurus-utils that would be appreciated :)

https://kentcdodds.com/blog/typescript-function-syntaxes#type-guards

**Comment by slorber** [comment]
Thanks 👍

**Pull request merged**